pkg/agent/attestor/node/node_test.go

```diff
@@ -373,32 +314,32 @@ func TestAttestor(t *testing.T) {
 			catalog.SetNodeAttestor(fakeagentcatalog.NodeAttestor("test", agentNA))
 			catalog.SetKeyManager(fakeagentcatalog.KeyManager(km))
 
-			// kick off the gRPC server serving the node API
-			serverAddr, serverDone := startNodeServer(t, tlsConfig, fakeNodeAPIConfig{
-				CACert:         caCert,
-				Attestor:       serverNA,
-				FailAttestCall: testCase.agentClient.failAttestAgent,
-			})
-			defer serverDone()
+			server := grpc.NewServer(grpc.Creds(credentials.NewTLS(tlsConfig)))
```

Comment:
Most of the changes in this file are to use an actual bundle and agent server for the tests. Currently, the test connects to a node api server but then configures bundle and agent test behavior via injected clients.

In removing the node api handler, the production code still needs to connect to something (via ServerAddr), so I opted to still keep a grpc server and instead of injecting clients, simply host the fake implementations of the bundle and agent services on the grpc server. Some of the test cases because harder to test then, like failures to send or recv on the stream, which is arguably low bang-for-buck.